### DISCUSSION TITLE
[Announcement] Xournal++ 1.1.1

### DISCUSSION THREAD

**Discussion by Technius** [xournalpp/xournalpp#3812] [General]
This is a new minor version of Xournal++ with many bug fixes and improvements from the community.

See the [changelog](https://github.com/xournalpp/xournalpp/blob/v1.1.1/CHANGELOG.md) for a full list of the changes.

Some notable changes:

* Change the edge panning behavior when an element is selected, fixing the absurdly fast edge pan speed bug.
* Fixed several crashing/freezing issues.
* Fixed several bugs involving stroke input and rendering.
* Fixed several bugs involving PDF rendering (in application) and export.
* Windows: fixed several bugs with fonts and the text tool.

<details><summary><b>Detailed commit log</b></summary>

* 4f113e9478a1908f3d1534f7a5b9763f7834d4d9 Release 1.1.1
* 10c189d39c29176f767d91bbf8fabde72d663a27 Update changelog in preparation for 1.1.1
* 7258d0bec14c4414eb20032dc295b9d561534be8 Fix issues with release script
* 773d063251b06570a10e9a681f00ec060fac16a0 Refactor BackgroundSelectDialogBase::layout
* e4e3ca56c77690d07fe33a579ef40a48ecb710e4 BackgroundSelectDialogBase: Fix layout logic
* 87ed1c39a74271665e1b0debdcdeaf90242ae377 Fix possible bug in edge pan timer destruction
* 3b1a2d7939f832328d17646644f5d826514b7d93 Update Authors lists
* c80e64bd3b84101084f0d13adb5fc5216a11596d Fix deadlock in SidebarPreviewPages
* 91ebfab120d35636bed554c926eaadeef3be55a9 Remove the .xopp~ backup file after a successful save [ #3399, #1498 ]
* 9f66a83e0063f3ffd5ba89d8eee7e862de4fbb3c fix indentation in build mac script
* 524f8a96c03fc9fc2153047bd8b760988d5f4feb Include cppunit for MacOs on Azure Pipelines
* ba0a547de7d29f6a5bbebaa82fa9907910b80089 Fixup GDK_PIXBUF_MODULE_FILE string replacement
* 1b03098e3c4944af84a272ce3b49cf2c5993389c Update macos blob and adapt loaders cache location
* 57654c0105ba058699c7869d485f095940d17e3e fix: Crash on startup with no recent files [ #3734 ]
* f961b04b42d7c5d0ddbb1e8f0b5785da0e9ffbdf Fix tool activation
* c1ae6c31268d828c8c827fcedae749099e630c3c Fix wrong version suffix in relevant files
* db7a16eec03b8c2b2fcb0db3cf193ff336cd99ba Update release helper script to set correct versions for Fedora and MacOS
* 95c50affabb682b91a8b214a34e21ebf8185ae74 Add version info in MacOS bundle information file
* be1dab569ef82a50960db2e3988f64d36f783be4 Update version info in Fedora build recipe
* 01737d1d47b3854c317b821410b84323a4502a63 Fix #3651 - remove pressure value on Highlighter strokes first point
* 29bee34dfee35d1a532cf90e0d069a6924c7b606 Fix .PDF detected as pdf
* 2780815e821cd8094f382f6a66e5055afb7e8382 fix sidebar page preview stuck
* e2a1f97327d94818cdecb19795308f8aed3177eb Set classic locale for PDF export link attributes (#3551)
* d88fb379473b582f32ee7565434f525413db1da9 Fixing typo (#3518)
* 0d8d375a1371ba3bdecf02de11e9cb39525ba8ff Update Debian Pipeline to Bullseye (#3547)
* 2ed7f70db8b1368a2f0023c4781c553b0f0ee555 Switch to FontConfig Backend for Cairo-Pango on Windows (#3534)
* c07d399b8be2f2c38e95b81f608c566d4ac44ad4 Add null check to device in InputEvent's handler (#3511)
* f76a2630f89d90801d7f3b3d31b63bc058278d00 Unreliable internationalization on Windows (#3426)
* 08923a32661cc0bee88c43939d64f666e9d47a9b Change Tool internals, increase static analysis. (#3465)
* 7bf1c91590cf1d8124323459d195b31441b0b33d Add unit tests for serialization in ToolEnum. (#3462)
* a81fe84a320bcbdf03d8b1591009cfb81c04156f Add const qualifier in serialization methods.
* 32a9ec402a6b45ab24bec66ee0cbef83a7dbc1ee Add const version of getter getElements().
* 9fa58be4c0cfde7882cbd53ec6dc1ae1a11936ce Add shared data of Poppler to Windows installer (#3477)
* a02baf53ed45f96a4b41506031390d44607c0a29 Disable CI triggers for installer pipeline
* 15dc736840e3b260c731844ff2a3a5ef7b2b40f9 Create pipeline for releases via comments
* 07264b02a89dbbad6fa6d26059adbed3cbb1495f Fix ToolType serialization bug. (#3467)
* c8d690c2d3eb1ebea008f433f6d7b0ba9968424c Switch from GMutex to std::mutex. (#3382)
* 412fc8e188a3f49a182feb472527278ad260cc7c ensure lgi loads Gtk 3
* 98808ff8a919b725ba5404574100ffa6da4c85e0 Fix memory leak in ExportDialog. (#3420)
* ccd9e9f95909b1b352dbc85d7af28b1aea249ce9 Enable pressure sensitivity of single dot strokes (#3344)
* f3f596bf5564516d324c34f72af9bd4ac248c3de Close dangling file handlers. (#3392)
* cbc29af9e8a9a38614f08438405dc3a61e653887 Bump development version to 1.1.1~dev
* cd72446d0eb9845c627e79cc4149b44f0ab724b7 Update tag trigger for release pipeline
* 71fe1f1680fb104d919385b3e070527a83958123 [skip ci] Update translation template
* 3742b7d1bb8b4af00a4f611ec70afe000996dd72 Release Helper Scripts (#3299)
* 3a77573cb62b481074a54a8d5f5a5adb555a7185 New Crowdin updates (#3301)
* 623d8654bacbc8db264f2cf9a76aead20be26889 Fix some non-English IMEs not working on Windows
* f59c3c4a400c9117aba719ee0011b952c532aa1d (Minor) Fix format specifiers to match types
* 466082db423f8a5abeb9bf1f46f2e52cf1a7c5ba Fix typo from 'orignal' to 'original'
* ab57ddda2489182039b4f267fc194dfe42bdccf8 Remove horizontal and vertical space compensation
* 9ceed9f0a8b34793ec456bfb813a07a82c0e9fc2 Remove console window flash at program startup on Windows
* ff4936e90f1d82784d68862033efd418ca6313dc Hide, not disable, contents tab when unavailable
* 82ade68b356e03f206edab8465e658cd715f3749 More natural zooming via ctrl+scroll and stepping.
* aaed673b72943d44f3ffce53cb8cf2231640f01f [skip ci] Update translation template
* d690146d1ebc63d996f080c618255c02176b4519 Fixed #3330: Allow characters after dot in filename (#3333)
* dad1ef7e93a6a0d337789d54bf93557711ecaac9 [skip ci] Update translation template
* e6bb8b797fa5ca66c55724bed5017863b7a0f36c Extend input method check for gcin (#3315)
* bd3ca75fe33ca1b4c02375e85f3976429fd2fc91 Fix inconsistent display of filled highlighter strokes
* 60ffda1d6c92ec16a1ceae56746139dff82baf5e Fix text placement snapping issue
* f41f7e3dfdec1659e16d227c2fb75819b061cfff Fixed 'PDF background missing' text showing (#3350)
* aca20186860d941195efd63cb3cd62068ae37907 Apply svgo to ui/pixmaps/*.svg
* 4192b838afe62184faec016508e8d47bd6537242 Replace masks by fill-opacity in ui/pixmaps/com.github.xournalpp.xournalpp.svg without altering the appearence
* 3fe13126e893f88f461cf48cf891f74654b73b63 [skip ci] Update translation template
* 76199be22c14040991129d70f915b939d48622cd Remove Placeholder ToolbarItem and fix drop index
* ca1c9da41c86ac72869f8b203352bee2324ac99a Fix off-by-one error in toolbar customization
* 4f8240210734e36df02d1e129dedbbe2ea30b1c3 [skip ci] Update translation template
* 7448364209f6399ad22eaa1cbddee496e8148324 Request dark theme from GTK when dark mode is requested in settings
* fffaa72e432a529cee7d343f0e337f491d98b431 [skip ci] Update translation template
* c09e9e389a4abc3fb30cd7d2b1c90c084a826102 Add preference settings for edge panning
* a10353dabeb8578f24376293727c5380f11c3cc5 Update settings.glade with new Glade version
* 51c98df1193e2bb3d1b9774bbfa6c9a788469556 Change selection panning behavior to use a timer
* 099b555c4688ad01867168720bc2f725932e1e7a Refactor EditSelection bounding box calculation
* 31a5164f76f3e3ba38281b3233e2ff52fa38a498 Simplify KeyboardInputHandler logic
* 18e18ddeb116ab77b596bd772e28006cb998bb20 Fix implicit type conversions in EditSelection
* e13cfbc3ca74e7f87041cd5b4ec1518f4f01137d Optimize GdkEventGuard
* c1adc54fad5dbd52971422f5dd90bd4a17675592 New Crowdin updates (#3300)
* f80f08b7af77ab4413036e42459413e561f4b4f7 Add Xournal++ pronunciation (#3231)
* 5b4bc64d849f02e13f5493035b6fb3af9cfa11a9 Update stroke recognizer to preserve line endpoints. (#3285)
* b20500b675349982ef64055cab4f22a2e46d6191 New Crowdin updates (#3244)
* c420c53c8cbf21819764802511494d869c0651c7 Remove using std::map from header (#3287)
* b030b67a2accd59e0e8ffa51534394ae0899c7df [skip ci] Update translation template
* f4102878f7ac7655ff7dde72c8a7dee63764544a Rebuild layer menu after un-/redone rename
* 77254ce25150981c21e3e283554f3670cf24a6c4 Remove unnecessary (erroneous) renames
* 0ab81f7fe078ad57d4e2951156e3f935f6d0e599 Remove X11 callback to flush the X11 context, which happens way too often.
* 1e39b2392c3c5fef43879d07b1943943fec6562e Optimized the calculation of stroke sizes
* 4b2d2dbdb543236c815cc8fc2e1e53234252ae7f Remove std::unordered_map to reduce caching issues / increase performance
* 97c20f62c5a5c16ce6dbb21ffd19240f4fe55961 Add Minimal Top toolbar
* d04020b9646a8878bce82407f6e16dcf5a46dbfc Add text tool, zoom slider and rename
* ef9983c2591d96f94267acbc2acecf20c937cdc4 Add minimal toolbar
* 5e454faf2ccdfc737bd1908b80dd793346e8a78c Update Portaudio to version 19.7 on MS Windows
* 949b20561a1291aae89046016e8b2f450c429f49 Remove outdated flags from Readme
* 4edec25954b1c610071e273c278ed8cd7ea45ba3 fixup compiler errors hiding behind flags
* 4abc8dfda2f573e29af7bd81f1e029e021facc58 generateConvert script: Change «, » to \\\". (#3271)
* 976cdce384b04f0f8aea130b320b6f84620ac3d5 Add full stroke repaint when using highligher and Fill
* 58d8534a45680fa353c46a2b0e0a58f1e2033058 Minimize the area repainted at each Stroke::drawSegmentTo
* 4ae824d96366fa6b20a962ca8afb718f952e8d13 [skip ci] Update translation template
* 55cbf477e82847f517f3617501b2d939ab44180e Fixed typos in Stabilizer's settings
* 5abfd934bfd24f3d0fa01e46c0f284bfb8fa521b [skip ci] Update translation template
* 3b10bb195d918cd280539312af17bd415155f78e Use npos instead of -1
* a5bc52f93e424073ba63f6424445e5e0b69564a5 Fix simple implicit type conversions
* b245355419b359f27268cd450e24348f659614c5 [skip ci] Update translation template
* cb9594d8419b39b3e6ed2be1fe23d327519edb36 Remove using std::* statements in header files
* 97b7460cee92744f0807116ab489e65279837b15 Avoid deprecated mimetypes tag
* ce1299e35e4d9340a6b4b8ed8254426b5a3974bd Update appdata screenshots

This list of changes was [auto generated](https://dev.azure.com/xournalpp/ce85de88-4b71-44a0-8a57-2707c0b9a6e7/_release?releaseId=499&_a=release-summary).</details>

<hr /><em>This discussion was created from the release <a href='https://github.com/xournalpp/xournalpp/releases/tag/v1.1.1'>Xournal++ 1.1.1</a>.</em>

**Comment by KangJieyu** [comment]
Mac版本的无法输入中文内容，建议修改
